pkg/bmc/client/bmclib/bmclib.go

```diff
@@ -0,0 +1,100 @@
+package bmclib
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/bmc-toolbox/bmclib"
+
+	"github.com/tinkerbell/rufio/pkg/bmc/client"
+)
+
+// BMCLibClient utilizes bmclib to interact with a baseboard management controller.
+type BMCLibClient struct {
+	client *bmclib.Client
+}
+
+func NewBMCLibClient() client.BMCClient {
+	return &BMCLibClient{}
+}
+
+// InitClient is used to initilize a bmclib client based on input host and credentials.
+func (bc *BMCLibClient) InitClient(hostIP, port, username, password string) {
```

Comment:
`client != nil` is an invariant of the `BMCLibClient` type. Without it `BMCLibClient` is virtually useless.

Instead, define an interface with the behaviors we want and allow consumers to inject constructs satisfying that behavior.  

Doing this will remove the need to check `bc.client == nil { return error }` everywhere.